### DISCUSSION TITLE
What is the trigger to start the syncing and how often it triggers?

### DISCUSSION THREAD

**Discussion by stauffenberg2020** [marcone/teslausb#839] [General]
Hi @marcone and all

I vaguely understand that teslausb works like this

USB is powered ON --> Gets connected to the wifi network  --> Wait for ARCHIVE_DELAY (say 20 seconds) --> Check (**_only once_** at the start) if there anything that needs to be synced --> If yes, then sync, If No, Then teslausb's syncing functions go to sleep mode and only the OTG is emulated for the car

The above is my understanding of how it works. Please correct me if I am wrong.

But what I am expecting is,
USB is powered ON --> Gets connected to the wifi network  --> Wait for ARCHIVE_DELAY (say 20 seconds) --> Check (**_continuously_**) if there anything that needs to be synced --> If yes, then sync (**_continuously_**), Else, dont sync thereby keeping the sync feature always ON as long as there is power supplied to the USB by the car

Can you please help me with this understanding?

**Comment by marcone** [comment]
Neither of these are how it works.

If the archive server is reachable on boot, it will sync immediately (without delay). After that, it will sync (with a delay) whenever it transitions from "not able to reach the archive server" to "able to reach the archive server".

**Reply by stauffenberg2020** [reply]
Hi @marcone . Thanks for your reply. I made an issue #840 explaining the need for a fix. Can you please let me know if you see these use cases completely unrealistic ?
